src/hotspot/cpu/s390/macroAssembler_s390.hpp

```diff
@@ -864,18 +864,17 @@ class MacroAssembler: public Assembler {
   //
 
   // Assert on CC (condition code in CPU state).
-  void asm_assert(bool check_equal, const char* msg, int id) PRODUCT_RETURN;
-  void asm_assert_low(const char *msg, int id) PRODUCT_RETURN;
-  void asm_assert_high(const char *msg, int id) PRODUCT_RETURN;
-  void asm_assert_eq(const char* msg, int id) { asm_assert(true, msg, id); }
-  void asm_assert_ne(const char* msg, int id) { asm_assert(false, msg, id); }
+  inline void asm_assert(branch_condition cond, const char* msg, int id, bool is_static=true);
 
-  void asm_assert_static(bool check_equal, const char* msg, int id) PRODUCT_RETURN;
+  // Assert if CC indicates "not equal" (check_equal==true) or "equal" (check_equal==false).
+  inline void asm_assert(bool check_equal, const char* msg, int id) {
```

Comment:
This will not work. Implementation of called method is not visible here. Need to move implementation to macroAssembler_s390.inline.hpp